cabal-install-solver/src/Distribution/Solver/Types/PkgConfigDb.hs

```diff
@@ -63,10 +68,14 @@ readPkgConfigDb verbosity progdb = handle ioErrorHandler $ do
     case mpkgConfig of
       Nothing             -> noPkgConfig "Cannot find pkg-config program"
       Just (pkgConfig, _) -> do
-        pkgList <- lines <$> getProgramOutput verbosity pkgConfig ["--list-all"]
+        -- To prevent malformed Unicode in the descriptions from crashing cabal,
+        -- read without interpreting any encoding first. (#9608)
+        pkgList <- LBS.split 10 <$> getProgramInvocationLBS verbosity (programInvocation pkgConfig ["--list-all"])
```

Comment:
I suggest we avoid the literal `10`, it confused me initially
```suggestion
        pkgList <- LBS.split (fromEnum '\n') <$> getProgramInvocationLBS verbosity (programInvocation pkgConfig ["--list-all"])
```

Reply:
It should be a `Word8`, so that'd become `fromIntegral (ord '\n')` which is quite verbose. But fair point, maybe one shouldn't assume extensive familiarity with ASCII from Cabal contributors :D

Reply:
I suppose you could also just add `-- 10 is ASCII for '\n'`, or `-- split on newlines (10 == fromEnum '\n')

Reply:
I made it the verbose version, and just added a newline ;)